### ISSUE TITLE
[4.0] fields javascript failure

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#26133]
### Steps to reproduce the issue
Try to create a new field of anything but text

### Expected result
when you select the field type then various options will appear appropriate to that field type

### Actual result
No change but there is a js error
```
Uncaught TypeError: Joomla.loadingLayer is not a function
    at window.typeHasChanged (index.php?option=com_fields&view=field&layout=edit&context=com_content.article:2102)
    at HTMLSelectElement.onchange (index.php?option=com_fields&view=field&layout=edit&context=com_content.article:337)
```

**Comment by C-Lodder** [comment]
Wherever `Joomla.loadingLayer()` is being callled from, it needs to be replaced with:

To **show** the loading icon:
```
document.body.appendChild(document.createElement('joomla-core-loader'));
```

To **hide** the loading icon:
```
const loaderElement = document.querySelector('joomla-core-loader');
loaderElement.parentNode.removeChild(loaderElement);
```

or:

```
document.body.removeChild(document.querySelector('joomla-core-loader'));
```

**Comment by ghost** [comment]
Closed as having Pull Request #26158
